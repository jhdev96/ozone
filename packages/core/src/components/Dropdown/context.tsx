import React from 'react';


interface ContextType {
  /** The name of the selected item. */
  selected?: string;
  /** A function that handles selection. */
  handleSelection?: (name: string) => void;
};

export const DropdownContext = React.createContext<ContextType>({
  selected: '',
  handleSelection: () => {}
});