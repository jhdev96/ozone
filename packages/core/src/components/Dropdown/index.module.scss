@import '../../scss/root';
@import '../../scss/button.scss';


.basic,
.filter {
  min-width: 121px;
  width: fit-content;
  align-self: center;

  &__button {
    width: 100%;
    display: inline-flex;
    justify-content: space-between;
    padding: 5px 8px;
    box-sizing: border-box;
  }
}

.basic { margin-right: 10px; }

.basic__button {
  @extend .button--basic;

  margin: 0;
  border-radius: 0;
  padding: 8px 10px;
  cursor: pointer;

  &--light {
    color: var(--black-core);
    background-color: #fff;
    border: 1px solid var(--gray-3);
  }

  &--dark {
    color: var(--gray-1);
    border: 1px solid var(--black-6);
    background-color: var(--black-4) !important;
  }
}

.dropdown__menu--basic {
  margin-top: -1px !important;
  border-radius: 0 !important;

  :first-child {
    margin-top: 0 !important;
  }

  .menu__item {
    padding: 0.569rem !important;

    &:hover {background: var(--blue-0);}
  }

  &--light {
    color: #353640;
    border: 1px solid #dfe7ee;
    background-color: #fff;
  }
  &--dark {
    color: var(--gray-core);
    border: 1px solid #2b343e;
    background-color: #383f47 !important;
  }
}

.filter__button {
  position: relative;
  z-index: 999;
  min-width: 50px;
  border-radius: 100px;
  font-size: 0.9em;
  cursor: pointer;

  &--light {
    color: var(--black-2);
    background-color: var(--gray-core);
  }

  &--dark {
    color: var(--gray-3);
    background-color: #32383f;
  }

  &--visible {
    border-radius: 10px 10px 0 0 !important;
  }
}

.dropdown__menu--filter {
  z-index: 1;
  border-radius: 0 0 10px 10px !important;

  &:last-child { padding-bottom: 7px; }

  &--light {
    color: var(--black-2);
    background-color: var(--gray-core);

    .menu__item:hover {
      color: var(--black-core);
      background-color: transparent;
    }
  }

  &--dark {
    color: var(--gray-3);
    background-color: var(--black-core);

    .menu__item:hover {
      color: #fff;
      background-color: transparent;
    }
  }
}

.dropdown__menu {
  visibility: hidden;
  position: absolute !important;
  min-width: 121px;
  font-size: 0.8em;
  border-radius: 4px;
  box-sizing: border-box;
  transition: 0.2s height;

  .menu__item {
    padding: 0.25rem 0.65rem;
    text-align: left;
    cursor: pointer;
    transition: 0.2s opacity;
  }
}

.selected__title {
  color: #25aae1;
  font-weight: 600;
  text-shadow: none;
}

.visible {
  visibility: visible;
}

.aboveTarget {
  .dropdown__menu--basic {
    bottom: 32px;
  }

  .filter__button {
    border-radius: 0 0 10px 10px !important;
  }

  .dropdown__menu--filter {
    bottom: 25px;
    padding-top: 5px;
    border-radius: 10px 10px 0 0 !important;
  }
}

.selected { color: #25aae1 !important; }