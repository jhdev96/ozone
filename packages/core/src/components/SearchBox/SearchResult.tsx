import React from 'react';
import classNames from '../../utils/classNames';
import styles from './index.module.scss';


interface SearchResultProps {
  /**
   * Assigns a name to the search result.
   * 
   * This is so the consumer of this component
   * can retrieve the name when the result is 
   * clicked.
   */
  name?: string;
  /**
   * Called when the result is clicked.
   */
  onClick?: (e: React.MouseEvent<HTMLDivElement>, name?: string) => void;
  /**
   * Provides additional styling to a search result.
   */
  className?: string;
  /**
   * A selector used in testing environments.
   */
  dataTestId?: string;
  /**
   * The result's content.
   */
  children?: React.ReactNode;
}

export default function SearchResult({
  name,
  onClick = () => {},
  className,
  dataTestId,
  children
}: SearchResultProps): JSX.Element 
{
  return (
    <div 
      className={classNames(className, styles.result)}
      onClick={(e) => onClick(e, name)}
      data-testid={dataTestId}
    >
      {children}
    </div>
  )
}

SearchResult.displayName = 'SearchResult';