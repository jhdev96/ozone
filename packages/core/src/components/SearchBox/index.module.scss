@import '../../scss/root';


.searchWrapper { position: relative; }

.searchContainer {
  position: relative;
  left: 0;
  width: fit-content;
  min-width: 500px;
  border: 1px solid transparent;
  border-radius: 20px !important;
  outline: none;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 5px 1px;
  transition: box-shadow 0.3s, height 0.4s;
  white-space: nowrap;
  box-sizing: border-box;

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.07) 0px 3px 15px 1px;
  }
}

.searchIcon {
  align-self: center;
  margin-left: 0.5em;
  margin-right: 0.2em;
  color: #adb3b9;

  svg {
    vertical-align: middle;
  }
}

.mainSearch {
  width: 100%;
  display: inline-flex;
}

.searchInput, 
.searchInput:focus {
  width: 100%;
  padding: 0.5rem 0.5rem;
  border: 0 !important;
  border-radius: 0 20px 20px 0;
  box-shadow: none;
  outline: 0 !important;
  font-size: 0.95em;

  &::placeholder {
    color: var(--gray-4);
  }
}

.searchResults {
  position: relative;
  z-index: 1;
  background-color: #fff;
  border-radius: 0 0 20px 20px;
  
  .result {
    padding: 0.5rem 0.75rem;
    cursor: pointer;

    &:last-child {
      border-radius: 0 0 20px 20px;
    }

    &:hover {
      background-color: var(--blue-0);
    }
  }
}

.searchFilter {
  margin-right: 0.25em;
  float: right;
  align-self: center;
}

.fill { min-width: 100%; }

.outline {
  box-shadow: none !important;
  border: 1px solid var(--gray-core);

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.04) 0px 2px 15px 1px;
  }
}

.focus {
  border: 1px solid var(--blue-core) !important; 
  box-shadow: 0 0 0 1px var(--blue-core) !important;
  transition: 0.3s box-shadow;

  &:hover {
    box-shadow: 0 0 0 1px var(--blue-core) !important;
  }
}

.light {
  background-color: #fff;

  .searchInput {
    background-color: #fff;
  }
}

.dark {
  background-color: var(--black-3);

  .searchInput {
    color: var(--gray-1);
    background-color: var(--black-3);
  }
}