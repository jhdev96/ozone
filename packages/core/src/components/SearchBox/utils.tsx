import React from 'react';
import styles from './index.module.scss';


export const toggleFocus = (e: React.FocusEvent<HTMLDivElement>) => {
  const classList = e.currentTarget.classList;
  
  if (classList.contains(styles.focus)) {
    classList.remove(styles.focus);
  } else {
      classList.add(styles.focus);
  }
}