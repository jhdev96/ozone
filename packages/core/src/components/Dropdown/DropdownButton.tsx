import React, { useContext } from 'react';
import classNames from '../../utils/classNames';
import { ThemeType } from '../../utils/theme';
import Colors from '@jhdev96/ozone-colors';
import { Arrow } from '@jhdev96/ozone-icons';
import { DropdownContext } from './context';
import styles from './index.module.scss';


interface DropdownButtonProps {
  /**
   * The type of dropdown being rendered.
   */
  variation?: "basic" | "filter";
  /**
   * Gives the button a light or dark theme.
   */
  theme?: ThemeType;
  /**
   * A function that is called when the button
   * is clicked.
   */
  onClick?: () => void;
  /**
   * Sets the direction of the arrow icon on
   * the button. This will be passed as a prop
   * to arrow icon.
   */
  arrowDirection?: "up" | "down";
  /**
   * This will be true when the dropdown is
   * visible.
   */
  showDropdown: boolean;
}

export default function DropdownButton({
  variation,
  theme,
  onClick,
  arrowDirection,
  showDropdown
}: DropdownButtonProps): JSX.Element 
{
  const {selected} = useContext(DropdownContext);

  return (
    <div
      onClick={onClick} 
      className={classNames(
        styles[`${variation}__button`], 
        styles[`${variation}__button--${theme}`],
        showDropdown && styles[`${variation}__button--visible`]
      )}
    >
      <span className={styles[`${variation}__title`]}>
        {selected}
      </span>
      <Arrow 
        direction={arrowDirection} 
        color={Colors.grayScale.BLACK2} 
      />
    </div>
  )
}