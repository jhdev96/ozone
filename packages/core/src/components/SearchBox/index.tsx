import React, { useState, useRef } from 'react';
import classNames from '../../utils/classNames';
import { ThemeType } from '../../utils/theme';
import { toggleFocus } from './utils';
import { Search as SearchIcon } from '@jhdev96/ozone-icons';
import Colors from '@jhdev96/ozone-colors';
import ClickManager from '../ClickManager';
import SearchResult from './SearchResult';
import styles from './index.module.scss';


interface SearchBoxProps {
  /** 
   * Adds a light border around the searchbox.
   */
  outline?: boolean;
  /** 
   * The text that is shown when the search
   * input is empty.
   */
  placeholder?: string;
  /** 
   * Gives the searchbox a light or dark appearance.
  */
  theme?: ThemeType;
  /** 
   * If true, the searchbox will occupty the full
   * width of its container.
   */
  fullWidth?: boolean;
  /**
   * Sets the `className` property of the searchbox.
   */ 
  className?: string;
  /** 
   * Function that is called when the search input's
   * value changes. 
   */
  onChange?: (query: string) => void;
  /**
   * Called when the searchbox gains focus.
   */
  onFocus?: (e: React.FocusEvent) => void;
  /** 
   * Called when the searchbox loses focus.
   */
  onBlur?: (e: React.FocusEvent) => void;
  /**
   * Called when a key is pressed down with the input focused.
   */
  onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  /**
   * Called when a key is released with the input focused.
   */
  onKeyUp?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  /** 
   * Called while a key is being pressed down with
   * the input focused.
   */
  onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  /** 
   * An element used for filtering search queries.
   */
  searchFilter?: React.ReactNode;
  /**
   * A selector used in testing environments.
   */
  dataTestId?: string;
  /** 
   * The markup provided to this prop
   * should be the search results that
   * are returned from a user's search
   * query.
   */
  children?: React.ReactNode;
} 

const SearchBox = React.forwardRef<HTMLInputElement, SearchBoxProps>(
  function SearchBox({
    outline       = false,
    placeholder   = "Search",
    theme         = "light",
    fullWidth     = false,
    className,
    onChange      = () => {},
    onFocus       = () => {},
    onBlur        = () => {},
    onKeyDown     = () => {},
    onKeyPress    = () => {},
    onKeyUp       = () => {},
    searchFilter,
    dataTestId,
    children}: SearchBoxProps,
    ref: React.ForwardedRef<HTMLInputElement>): JSX.Element 
  {
    const [results, setResults] = useState<React.ReactNode>(children);
    const wrapperRef = useRef<HTMLDivElement>();

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
      setResults(children);
      onChange(e.target.value);
    }

    const handleFocus = (e: React.FocusEvent<HTMLDivElement>) => {
      toggleFocus(e);
      onFocus(e);
    }

    const handleBlur = (e: React.FocusEvent<HTMLDivElement>) => {
      toggleFocus(e);
      onBlur(e);
    }

    return (
      <ClickManager 
        ref={wrapperRef} 
        className={styles.searchWrapper} 
        onClick={() => results && setResults(null)}
      >
        <div 
          onFocus={(e) => handleFocus(e)}
          onBlur={(e) => handleBlur(e)}
          className={classNames(
            styles.searchContainer,
            styles[theme],
            outline   && styles.outline,
            fullWidth && styles.fill,
            className
          )}
          data-testid={dataTestId}
        >
          <div className={styles.mainSearch}>
            <span className={styles.searchIcon}>
              <SearchIcon 
                color={
                  theme === 'dark' 
                  ? Colors.grayScale.GRAY4  
                  : Colors.grayScale.BLACK2
                } 
              />
            </span>
            <input 
              type="text" 
              className={styles.searchInput} 
              placeholder={placeholder} 
              onChange={handleChange}
              onKeyDown={onKeyDown}
              onKeyPress={onKeyPress}
              onKeyUp={onKeyUp}
              autoComplete="off"
              ref={ref}
            />
            <span className={styles.searchFilter}>
              {searchFilter}
            </span>
          </div>
          <div className={styles.searchResults}>
            {results && children}
          </div>
        </div>
      </ClickManager>
    );
  }
)

export { SearchResult };
export default SearchBox;

SearchBox.displayName = 'SearchBox';