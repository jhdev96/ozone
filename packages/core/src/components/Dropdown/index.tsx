import React, { useState, useRef, LegacyRef } from 'react';
import classNames from '../../utils/classNames';
import { ThemeType } from '../../utils/theme';
import { DropdownContext } from './context';
import DropdownButton from './DropdownButton';
import Manager from './Manager';
import Item from './Item';
import styles from './index.module.scss';


interface DropdownProps {
  /** 
   * Sets the theme for the dropdown. 
   */
  theme?: ThemeType;
  /** 
   * Main classnames for dropdowns. 
   */
  variation?: "basic" | "filter";
  /** 
   * When true, this will display 
   * the selected option as the new title. 
   */
  saveSelection?: Boolean;
  /** 
   * Give the dropdown a default option when rendered. 
   */
  defaultValue?: string;
  /** 
   * The function to be invoked when a selection is made. 
   * A `selectedItem` parameter should be given to the
   * callback so that when an option is selected, its
   * value is received by the dropdown's parent element.
   */
  onSelection?: (selectedItem: string) => void;
  /**
   * A selector used in testing environments.
   */
  dataTestId?: string;
  /**
   * The dropdown menu items.
   */
  children?: React.ReactNode;
}

export default function Dropdown({
  theme           = "light", 
  variation       = "basic",
  saveSelection,
  defaultValue    = "Dropdown",
  onSelection     = () => {},
  dataTestId,
  children
}: DropdownProps): JSX.Element 
{
  const [selected, setSelected] = useState<string>(defaultValue);
  const [visible, setVisible] = useState(false);
  const [aboveTarget, setAboveTarget] = useState(false);
  const wrapperRef = useRef<HTMLDivElement>();
  const menuRef = useRef<HTMLDivElement>();

  const toggleDropdown = () => {
    setVisible(!visible);
    setAboveTarget(false);
  }

  const handleSelection = (name: string) => {
    /** 
     * display the selected option from the dropdown menu 
     * if the `saveSelection` prop is set to true.
     */
    if (saveSelection) setSelected(name); 
    
    name && onSelection(name);
    toggleDropdown();
  }
       
  return (
    <Manager
      clickRef={wrapperRef}
      overflowRef={menuRef}
      className={classNames(
        styles[variation],
        aboveTarget && styles.aboveTarget
      )}
      onClick={() => {visible && toggleDropdown()}}
      setStateFn={setAboveTarget}
      showDropdown={visible}
    >
      <DropdownContext.Provider value={{selected, handleSelection}}>
        <DropdownButton
          variation={variation}
          theme={theme}
          onClick={toggleDropdown}
          showDropdown={visible}
          arrowDirection={aboveTarget ? "up" : "down"}
        />
        <div 
          className={classNames(
            styles['dropdown__menu'],
            styles[`dropdown__menu--${variation}`],
            styles[`dropdown__menu--${variation}--${theme}`],
            visible && styles.visible
          )}
          ref={menuRef as LegacyRef<HTMLDivElement>}
          data-testid={dataTestId}
        >
          {children}
        </div>
      </DropdownContext.Provider>
    </Manager>
  )
}

Dropdown.displayName = 'Dropdown';
Dropdown.Item = Item;