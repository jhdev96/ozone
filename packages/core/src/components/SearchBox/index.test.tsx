import React from "react";
import { 
  render, 
  screen, 
  fireEvent
} from "@testing-library/react"; 
import Dropdown from "../Dropdown";
import SearchBox, { SearchResult } from ".";


describe('<Searchbox />', () => {
  it('renders a basic searchbox with a placeholder', () => {

    render(<SearchBox dataTestId="test" />);

    const searchInput = document.querySelector('.searchInput');

    expect(searchInput!.getAttribute('placeholder')).toBe('Search');
  });

  it('renders a search icon inside the searchbox', () => {
    render(<SearchBox dataTestId="test" />);

    expect(screen.getByTestId('test'))
      .toMatchSnapshot();
  });

  it('renders a searchbox with an outline', () => {
    const {container} = render(<SearchBox outline />);

    expect(container).toMatchSnapshot();
  });

  it('renders a searchbox with a filter', () => {
    render(
      <SearchBox
        searchFilter={
          <Dropdown variation="basic" dataTestId="test">
            <Dropdown.Item>A</Dropdown.Item>
            <Dropdown.Item>B</Dropdown.Item>
            <Dropdown.Item>C</Dropdown.Item>
          </Dropdown>
        } 
      />
    );

    expect(screen.getByTestId('test'))
      .toBeInTheDocument();
  });

  it('changes the search input\'s value when a query is entered', () => {
    render(<SearchBox />);

    const searchInput = document.querySelector('.searchInput') as HTMLInputElement;

    fireEvent.change(searchInput!, {
      target: {value: 'hello'}
    });

    expect(searchInput.value).toBe('hello');
  });

  it('calls `onChange` with the input\'s value', () => {
    const onChange = jest.fn();

    render(<SearchBox onChange={onChange} />);

    const searchInput = document.querySelector('.searchInput');

    fireEvent.change(searchInput!, {
      target: {value: 'hello'}
    });

    expect(onChange).toBeCalledWith('hello');
  });

  it('adds a special css class when the searchbox is focused', () => {
    render(<SearchBox />);

    const wrapper = document.querySelector('.searchContainer');
    fireEvent.focus(wrapper!);

    expect(wrapper).toHaveClass('focus');
  });

  it('calls `onFocus` when the searchbox is focused', () => {
    const onFocus = jest.fn();

    render(<SearchBox onFocus={onFocus} />);

    const wrapper = document.querySelector('.searchContainer');
    fireEvent.focus(wrapper!);

    expect(onFocus).toBeCalled();
  });

  it('calls `onBlur` when the searchbox loses focus', () => {
    const onBlur = jest.fn();

    render(<SearchBox onBlur={onBlur} />);

    const wrapper = document.querySelector('.searchContainer');
    fireEvent.blur(wrapper!);

    expect(onBlur).toBeCalled();
  });

  it('calls `onKeyDown` when a keyDown event fires', () => {
    const onKeyDown = jest.fn();

    render(<SearchBox onKeyDown={onKeyDown} />);

    const searchInput = document.querySelector('.searchInput');
    fireEvent.keyDown(searchInput!);

    expect(onKeyDown).toBeCalled();
  });

  it('renders a searchbox with results', () => {
    render(
      <SearchBox>
        <SearchResult>apple</SearchResult>
        <SearchResult>banana</SearchResult>
        <SearchResult>orange</SearchResult>
      </SearchBox>
    );

    const results = document.querySelector('.searchResults');

    expect(results!.children).toHaveLength(3);
  });
});

describe('<SearchResult />', () => {
  it('renders a search result with text', () => {
    render(
      <SearchResult dataTestId="test">
        text
      </SearchResult>
    );

    expect(screen.getByTestId('test'))
      .toHaveTextContent('text');
  });

  it('matches snapshot', () => {
    const {container} = render(
      <SearchResult>text</SearchResult>
    );

    expect(container).toMatchSnapshot();
  });

  it('calls `onClick` when the search result is clicked', () =>{
    const onClick = jest.fn();
    
    render(
      <SearchResult 
        dataTestId="test"
        onClick={onClick}
      >
        text
      </SearchResult>
    );

    fireEvent.click(screen.getByTestId('test'));

    expect(onClick).toBeCalled();
  });
});