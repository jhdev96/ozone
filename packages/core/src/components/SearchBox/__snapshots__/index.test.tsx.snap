// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> matches snapshot 1`] = `
<div>
  <div
    class="result"
  >
    text
  </div>
</div>
`;

exports[`<Searchbox /> renders a search icon inside the searchbox 1`] = `
<div
  class="searchContainer light"
  data-testid="test"
>
  <div
    class="mainSearch"
  >
    <span
      class="searchIcon"
    >
      <svg
        height="14.6"
        viewBox="0 0 14.6 16.6"
        width="12.6"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          id="Ellipse_1"
          stroke="#7F808A"
          stroke-width="2.5"
        >
          <circle
            cx="6"
            cy="6"
            r="6"
            stroke="none"
          />
          <circle
            cx="6"
            cy="6"
            fill="none"
            r="4.75"
          />
        </g>
        <line
          fill="none"
          id="Line_5"
          stroke="#7F808A"
          stroke-linecap="round"
          stroke-width="3"
          transform="translate(9.5 10.5)"
          x2="3"
          y2="4"
        />
      </svg>
    </span>
    <input
      autocomplete="off"
      class="searchInput"
      placeholder="Search"
      type="text"
    />
    <span
      class="searchFilter"
    />
  </div>
  <div
    class="searchResults"
  />
</div>
`;

exports[`<Searchbox /> renders a searchbox with an outline 1`] = `
<div>
  <div
    class="searchWrapper"
  >
    <div
      class="searchContainer light outline"
    >
      <div
        class="mainSearch"
      >
        <span
          class="searchIcon"
        >
          <svg
            height="14.6"
            viewBox="0 0 14.6 16.6"
            width="12.6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              id="Ellipse_1"
              stroke="#7F808A"
              stroke-width="2.5"
            >
              <circle
                cx="6"
                cy="6"
                r="6"
                stroke="none"
              />
              <circle
                cx="6"
                cy="6"
                fill="none"
                r="4.75"
              />
            </g>
            <line
              fill="none"
              id="Line_5"
              stroke="#7F808A"
              stroke-linecap="round"
              stroke-width="3"
              transform="translate(9.5 10.5)"
              x2="3"
              y2="4"
            />
          </svg>
        </span>
        <input
          autocomplete="off"
          class="searchInput"
          placeholder="Search"
          type="text"
        />
        <span
          class="searchFilter"
        />
      </div>
      <div
        class="searchResults"
      />
    </div>
  </div>
</div>
`;
